babel-plugin-optimize-obj-str/src/index.js

```diff
@@ -0,0 +1,119 @@
+module.exports = function babelPluginOptimizeObjstr(babel) {
+	const { types: t } = babel;
+
+	/**
+	 * Fails on strict mode when encountering an unoptimizable case.
+	 */
+	function throwUnoptimizable(path, state, message) {
+		if (state.opts.strict) {
+			throw path.buildCodeFrameError(message);
+		}
+	}
+
+	/**
+	 * Decontextualizes a node for comparison with a different node, irrespective
+	 * of its location in source or surrounding comments.
+	 */
+	function decontextualize(node) {
+		const clean = { ...node };
+		delete clean.extra;
+		delete clean.loc;
+		delete clean.start;
+		delete clean.end;
+		delete clean.range;
+		delete clean.innerComments;
+		delete clean.trailingComments;
+		delete clean.leadingComments;
```

Comment:
Is it possible to build a new `clean` object with explicit properties instead of deleting properties?

Reply:
I don't think explicit properties are property since field sets vary by node type. Alternatively:

```suggestion
		const {
			extra: unusedExtra,
			loc: unusedLoc,
			start: unusedStart,
			end: unusedEnd,
			range: unusedRange,
			innerComments: unusedInnerComments,
			trailingComments: unusedTrailingComments,
			leadingComments: unusedLeadingComments,
			...clean
		} = node;
```

Reply:
I don't think explicit properties are possible since field sets vary by node type. Alternatively:

```suggestion
		const {
			extra: unusedExtra,
			loc: unusedLoc,
			start: unusedStart,
			end: unusedEnd,
			range: unusedRange,
			innerComments: unusedInnerComments,
			trailingComments: unusedTrailingComments,
			leadingComments: unusedLeadingComments,
			...clean
		} = node;
```

Reply:
Gotcha, makes sense then :)